src/geometry/euclidean/ceva_theorem.lean

```diff
@@ -0,0 +1,325 @@
+/-
+Copyright (c) 2021 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import tactic.linarith
+import tactic.fin_cases
+import tactic.norm_num
+import geometry.euclidean
+import linear_algebra.affine_space.basis
+import tactic.basic
+import analysis.normed_space.add_torsor_bases
+import data.equiv.basic
+import group_theory.perm.list
+
+/-!
+# Ceva's Theorem
+
+This file (although could be merged to another file later) proves Ceva's Theorem in Euclidean
+Geometry following the barycentric coordinate proof. For a reference,
+see https://en.wikipedia.org/wiki/Ceva%27s_theorem.
+
+Because of the currently generality that `interior_convex_hull_aff_basis` is stated in, we prove
+Ceva's theorem in it's current form. This will need to be upgraded to the usual setting of
+Euclidean Geometry results after the affinity refactor.
+
+**This is also a Problem 61 on Freek's list**
+-/
+
+variables {V : Type*} [inner_product_space ℝ V] [normed_space ℝ V]
+include V
+
+open affine finite_dimensional finset equiv affine_basis fintype basis
+
+open_locale euclidean_geometry big_operators
+
+lemma affine_basis.vsub_eq_coord_smul_sum (O : V) {ι : Type*} [fintype ι]
+  (S : affine_basis ι ℝ V) : ∀ X : V, X -ᵥ O = ∑ (i : ι), S.coord i O • (X -ᵥ S.points i) :=
+begin
+  set w : ι → ℝ := λ i, S.coord i O with hw,
+  have hs := sum_coord_apply_eq_one S O,
+  intro X,
+  have h₄ := weighted_vsub_of_point_vadd_eq_of_sum_eq_one univ w S.points hs X O,
+  rw vadd_eq_vadd_iff_sub_eq_vsub at h₄,
+  rw ← h₄,
+  have h₅ : (univ.weighted_vsub_of_point S.points O) w = (0 : V),
+  { suffices h : univ.affine_combination S.points w = O,
+    { rw affine_combination_eq_weighted_vsub_of_point_vadd_of_sum_eq_one _ w S.points hs O at h,
+      symmetry' at h,
+      rw eq_vadd_iff_vsub_eq at h,
+      rw ← h,
+      exact vsub_self O},
+    convert affine_combination_coord_eq_self S O },
+  rw h₅,
+  simp only [vsub_eq_sub, zero_sub, weighted_vsub_of_point_apply, neg_eq_iff_add_eq_zero,
+      ← sum_add_distrib, smul_sub, smul_sub],
+  norm_num,
+end
```

Comment:
See the distributivity lemmas I've proposed in #10786: this lemma showed up that some more general, simpler lemmas were missing. I think this specific case should follow more or less immediately from `sum_smul_vsub_const_eq_vsub_affine_combination` in that PR.

Reply:
After merging master into this branch to get those distributivity lemmas, this can be simplified to:

```suggestion
lemma affine_basis.vsub_eq_coord_smul_sum (O : V) {ι : Type*} [fintype ι]
  (S : affine_basis ι ℝ V) (X : V) : X -ᵥ O = ∑ (i : ι), S.coord i O • (X -ᵥ S.points i) :=
begin
  convert (finset.univ.sum_smul_const_vsub_eq_vsub_affine_combination (λ i, S.coord i O)
    S.points X (S.sum_coord_apply_eq_one O)).symm,
  exact (S.affine_combination_coord_eq_self _).symm
end
```